### DISCUSSION TITLE
May I ask, in the code display area on GitHub, on the right side, there is a listed display of the methods, properties, and classes of the code (which can be extracted for any language). What library is this based on?

### DISCUSSION THREAD

**Discussion by liaodalin19903** [community/community#131024] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

May I ask, in the code display area on GitHub, on the right side, there is a listed display of the methods, properties, and classes of the code (which can be extracted for any language). What library is this based on?
![image](https://github.com/community/community/assets/12727582/6c50aa49-0177-4b1b-9b95-5a705dc855bd)

What library is this based on?

**Comment by DaviZCodes** [comment, marked as answer]
That code display area is called the _Symbols_ panel. The library or tool that does this is called _Tree Sitter_ https://github.com/tree-sitter. 
This information about the panel is from: https://docs.github.com/en/enterprise-cloud@latest/search-github/github-code-search/understanding-github-code-search-syntax#symbol-qualifier.

I hope this helps. If it does, please mark my answer as the answer.
